src/expr.rs

```diff
@@ -624,11 +624,20 @@ ast_enum! {
 
 #[cfg(any(feature = "parsing", feature = "printing"))]
 #[cfg(feature = "full")]
-fn arm_requires_comma(arm: &Arm) -> bool {
-    if let ExprKind::Block(ExprBlock { unsafety: Unsafety::Normal, .. }) = arm.body.node {
-        false
-    } else {
-        true
+fn arm_expr_requires_comma(expr: &Expr) -> bool {
```

Comment:
I don't think this is correct -- the rule for expressions that need a semicolon in order to be a statement is different from the rule for expressions that need a comma in order to be a match arm. The link is for the former. I haven't tried it but does this change make syn erroneously parse [this code](https://play.rust-lang.org/?gist=5cf023825e7eeffa1ff5ac42f293c60a)?

```rust
fn main() {
    match 0 {
        1 => loop {}
        _ => {}
    }
}
```

Reply:
Oh this changed in nightly! Never mind.

Reply:
Indeed!